### ISSUE TITLE
React Native's initial project doesn't run on IOS.

### ISSUE THREAD

**Issue by sumanbhattarai** [facebook/react-native#31249]
## Description
I created a new project using the command `npx react-native init MyApp --template react-native-template-typescript`.
When i run` npx react-native run-ios`, it gives me below error.

```
** BUILD FAILED **
The following build commands failed:
        PhaseScriptExecution [CP-User]\ Generate\ Specs /Users/macbookpro/Library/Developer/Xcode/DerivedData/MyApp-gvadqhjqjqogdegfwqsahxwqnkps/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/FBReactNativeSpec.build/Script-8C01FEA837F8EC37C13D05D97D3F6C1B.sh
(1 failure)
```

## React Native version:
System:
    OS: macOS 11.2.3
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Memory: 58.32 MB / 8.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.16.0 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.14.11 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 14.0.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.64.0 => 0.64.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

**Comment by Alimahmood7** [comment]
I am also facing sort of same error 
<img width="1076" alt="Screen Shot 2021-03-26 at 11 28 56 PM" src="https://user-images.githubusercontent.com/17236308/112686087-4728e380-8e97-11eb-8fa0-7b27b19fc72d.png">

**Comment by imkrunal** [comment]
I am facing the same issue with `yarn ios` but if I build the app from Xcode it works.

**Comment by sankalp98** [comment]
I am facing the same issue. React native 0.64 and Xcode 12.4

**Comment by Alimahmood7** [comment]
> I am facing the same issue with `yarn ios` but if I build the app from Xcode it works.

Can you please tell me the steps to build app in xcode ?

**Comment by imkrunal** [comment]
> > I am facing the same issue with `yarn ios` but if I build the app from Xcode it works.
> 
> Can you please tell me the steps to build app in xcode ?

1. Open your ios folder in Xcode
2. In the top bar select your emulator and click on the build/play icon.

**Comment by sumanbhattarai** [comment]
@imkrunal no luck. Builing the app from Xcode gives me the same error.

**Comment by ugglr** [comment]
Same issue. Works with Xcode. 👀

**Comment by imkrunal** [comment]
> @imkrunal no luck. Building the app from Xcode gives me the same error.

Make sure you don't have space in the folder names. Check this comment.

https://github.com/facebook/react-native/issues/31259#issuecomment-808899927

**Comment by sumanbhattarai** [comment]
@imkrunal yes, I checked it. I had no space in the folder names.

**Comment by sumanbhattarai** [comment]
> > @imkrunal no luck. Building the app from Xcode gives me the same error.
> 
> Make sure you don't have space in the folder names. Check this comment.
> 
> [#31259 (comment)](https://github.com/facebook/react-native/issues/31259#issuecomment-808899927)
My path is:` /Users/macbookpro/Desktop/MyApp`

**Comment by sumanbhattarai** [comment]
Removing 
```
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
```
from `~/.bash_profile helped to solve this issue.

**Comment by alexduros** [comment]
Actually, this helps to make this stuff work but there is definitely an issue with `nvm` along with the exported env variables in the iOS build phase, espacially the `PATH` which makes the `command which node` differs from you login environnment, and therefore makes `nvm` raise an error in the build. 

I finally decided to do the same as @sumanbhattarai and rely on `nodenv` to make it work. 

Is there someone aware of how the environment variables are set by Xcode for the build phases and how to customize/fix it ?

**Comment by hlynbech** [comment]
We had the same issue when building on a build server with fastlane, when fastlane is started from a shell that has NVM environment variables set.
Removing the NVM variables from the shell (in this case in .zshrc) makes the build work in fastlane.
The build never had issues when building from the Xcode UI.

**Comment by hlynbech** [comment]
> [...] Commenting out all the code from `project/node_modules/react-native/scripts/find-node.sh`
> helped to solve this issue.

@sumanbhattarai Why did you close this issue, I know that disabling NVM and removing code from find-node.sh resolves it, but then there must be a bug in find-node.sh which is under the control of the react-native project.

It seems that some effort is going on in react-native to support also NVM, so this bug must be fixed, right? Or is it resolved in another issue?

**Comment by sumanbhattarai** [comment]
@hlynbech okay, i have reopened it.

**Comment by williamhaley** [comment]
This patch got things working for me with `nodenv` on macOS 11.3.1 with Xcode 12.5

```diff
diff --git a/node_modules/react-native/scripts/find-node.sh b/node_modules/react-native/scripts/find-node.sh
index c111e70..001de42 100755
--- a/node_modules/react-native/scripts/find-node.sh
+++ b/node_modules/react-native/scripts/find-node.sh
@@ -17,9 +17,9 @@ fi
 
 # Set up the nodenv node version manager if present
 if [[ -x "$HOME/.nodenv/bin/nodenv" ]]; then
-  eval "$("$HOME/.nodenv/bin/nodenv" init -)"
+  eval "$("$HOME/.nodenv/bin/nodenv" init --no-rehash -)"
 elif [[ -x "$(command -v brew)" && -x "$(brew --prefix nodenv)/bin/nodenv" ]]; then
-  eval "$("$(brew --prefix nodenv)/bin/nodenv" init -)"
+  eval "$("$(brew --prefix nodenv)/bin/nodenv" init --no-rehash -)"
 fi
 
 # Set up the ndenv of anyenv if preset

```

This is specific to `nodenv` mostly, but I adding a bunch of logging to `find-node.sh`.  _Without_ the `--no-rehash` flag, `nodenv` apparently attemps to `rehash` https://github.com/nodenv/nodenv/blob/master/libexec/nodenv-init#L104 when the command is eventually evaluated.

I didn't look into all the details, but this seemed to be the ultimate error.

```
nodenv: cannot rehash: /Users/will/.nodenv/shims/.nodenv-shim exists
```

Here's my full (more or less) Build Phases log for `[CP-User] Generate Specs`.

```
PhaseScriptExecution [CP-User]\ Generate\ Specs /Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Script-B4E323DB11FD523AF1F108EDFA5396CE.sh (in target 'FBReactNativeSpec' from project 'Pods')
    cd /Users/will/dev/myapp/ios/Pods
    export ACTION\=build
    export AD_HOC_CODE_SIGNING_ALLOWED\=NO
    export ALTERNATE_GROUP\=staff
    export ALTERNATE_MODE\=u+w,go-w,a+rX
    export ALTERNATE_OWNER\=will
    export ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES\=NO
    export ALWAYS_SEARCH_USER_PATHS\=NO
    export ALWAYS_USE_SEPARATE_HEADERMAPS\=NO
    export APPLE_INTERNAL_DEVELOPER_DIR\=/AppleInternal/Developer
    export APPLE_INTERNAL_DIR\=/AppleInternal
    export APPLE_INTERNAL_DOCUMENTATION_DIR\=/AppleInternal/Documentation
    export APPLE_INTERNAL_LIBRARY_DIR\=/AppleInternal/Library
    export APPLE_INTERNAL_TOOLS\=/AppleInternal/Developer/Tools
    export APPLICATION_EXTENSION_API_ONLY\=NO
    export APPLY_RULES_IN_COPY_FILES\=NO
    export APPLY_RULES_IN_COPY_HEADERS\=NO
    export ARCHS\=arm64
    export ARCHS_STANDARD\=arm64\ armv7
    export ARCHS_STANDARD_32_64_BIT\=armv7\ arm64
    export ARCHS_STANDARD_32_BIT\=armv7
    export ARCHS_STANDARD_64_BIT\=arm64
    export ARCHS_STANDARD_INCLUDING_64_BIT\=arm64\ armv7
    export ARCHS_UNIVERSAL_IPHONE_OS\=armv7\ arm64
    export ASSETCATALOG_FILTER_FOR_DEVICE_MODEL\=iPhone10,1
    export ASSETCATALOG_FILTER_FOR_DEVICE_OS_VERSION\=14.4.2
    export AVAILABLE_PLATFORMS\=appletvos\ appletvsimulator\ iphoneos\ iphonesimulator\ macosx\ watchos\ watchsimulator
    export BITCODE_GENERATION_MODE\=marker
    export BUILD_ACTIVE_RESOURCES_ONLY\=YES
    export BUILD_COMPONENTS\=headers\ build
    export BUILD_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products
    export BUILD_LIBRARY_FOR_DISTRIBUTION\=NO
    export BUILD_ROOT\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products
    export BUILD_STYLE\=
    export BUILD_VARIANTS\=normal
    export BUILT_PRODUCTS_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec
    export BUNDLE_CONTENTS_FOLDER_PATH_deep\=Contents/
    export BUNDLE_EXECUTABLE_FOLDER_NAME_deep\=MacOS
    export BUNDLE_FORMAT\=shallow
    export BUNDLE_FRAMEWORKS_FOLDER_PATH\=Frameworks
    export BUNDLE_PLUGINS_FOLDER_PATH\=PlugIns
    export BUNDLE_PRIVATE_HEADERS_FOLDER_PATH\=PrivateHeaders
    export BUNDLE_PUBLIC_HEADERS_FOLDER_PATH\=Headers
    export CACHE_ROOT\=/var/folders/gf/1_j5nggn6hj4yd0_0b7vg3y80000gn/C/com.apple.DeveloperTools/12.5-12E262/Xcode
    export CCHROOT\=/var/folders/gf/1_j5nggn6hj4yd0_0b7vg3y80000gn/C/com.apple.DeveloperTools/12.5-12E262/Xcode
    export CHMOD\=/bin/chmod
    export CHOWN\=/usr/sbin/chown
    export CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED\=YES
    export CLANG_ANALYZER_NONNULL\=YES
    export CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION\=YES_AGGRESSIVE
    export CLANG_CXX_LANGUAGE_STANDARD\=c++14
    export CLANG_CXX_LIBRARY\=libc++
    export CLANG_ENABLE_MODULES\=YES
    export CLANG_ENABLE_MODULE_DEBUGGING\=NO
    export CLANG_ENABLE_OBJC_ARC\=YES
    export CLANG_ENABLE_OBJC_WEAK\=NO
    export CLANG_MODULES_BUILD_SESSION_FILE\=/Users/will/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation
    export CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING\=YES
    export CLANG_WARN_BOOL_CONVERSION\=YES
    export CLANG_WARN_COMMA\=YES
    export CLANG_WARN_CONSTANT_CONVERSION\=YES
    export CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS\=YES
    export CLANG_WARN_DIRECT_OBJC_ISA_USAGE\=YES_ERROR
    export CLANG_WARN_DOCUMENTATION_COMMENTS\=YES
    export CLANG_WARN_EMPTY_BODY\=YES
    export CLANG_WARN_ENUM_CONVERSION\=YES
    export CLANG_WARN_INFINITE_RECURSION\=YES
    export CLANG_WARN_INT_CONVERSION\=YES
    export CLANG_WARN_NON_LITERAL_NULL_CONVERSION\=YES
    export CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF\=YES
    export CLANG_WARN_OBJC_LITERAL_CONVERSION\=YES
    export CLANG_WARN_OBJC_ROOT_CLASS\=YES_ERROR
    export CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER\=NO
    export CLANG_WARN_RANGE_LOOP_ANALYSIS\=YES
    export CLANG_WARN_STRICT_PROTOTYPES\=YES
    export CLANG_WARN_SUSPICIOUS_MOVE\=YES
    export CLANG_WARN_UNGUARDED_AVAILABILITY\=YES_AGGRESSIVE
    export CLANG_WARN_UNREACHABLE_CODE\=YES
    export CLANG_WARN__DUPLICATE_METHOD_MATCH\=YES
    export CLASS_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/JavaClasses
    export CLEAN_PRECOMPS\=YES
    export CLONE_HEADERS\=NO
    export CODESIGNING_FOLDER_PATH\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec/libFBReactNativeSpec.a
    export CODE_SIGNING_ALLOWED\=NO
    export CODE_SIGNING_REQUIRED\=YES
    export CODE_SIGN_CONTEXT_CLASS\=XCiPhoneOSCodeSignContext
    export CODE_SIGN_INJECT_BASE_ENTITLEMENTS\=YES
    export COLOR_DIAGNOSTICS\=NO
    export COMBINE_HIDPI_IMAGES\=NO
    export COMPILER_INDEX_STORE_ENABLE\=Default
    export COMPOSITE_SDK_DIRS\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/CompositeSDKs
    export COMPRESS_PNG_FILES\=YES
    export CONFIGURATION\=Debug
    export CONFIGURATION_BUILD_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec
    export CONFIGURATION_TEMP_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos
    export COPYING_PRESERVES_HFS_DATA\=NO
    export COPY_HEADERS_RUN_UNIFDEF\=NO
    export COPY_PHASE_STRIP\=NO
    export COPY_RESOURCES_FROM_STATIC_FRAMEWORKS\=YES
    export CORRESPONDING_SIMULATOR_PLATFORM_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform
    export CORRESPONDING_SIMULATOR_PLATFORM_NAME\=iphonesimulator
    export CORRESPONDING_SIMULATOR_SDK_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.5.sdk
    export CORRESPONDING_SIMULATOR_SDK_NAME\=iphonesimulator14.5
    export CP\=/bin/cp
    export CREATE_INFOPLIST_SECTION_IN_BINARY\=NO
    export CURRENT_ARCH\=undefined_arch
    export CURRENT_VARIANT\=normal
    export DEAD_CODE_STRIPPING\=YES
    export DEBUGGING_SYMBOLS\=YES
    export DEBUG_INFORMATION_FORMAT\=dwarf
    export DEFAULT_COMPILER\=com.apple.compilers.llvm.clang.1_0
    export DEFAULT_DEXT_INSTALL_PATH\=/System/Library/DriverExtensions
    export DEFAULT_KEXT_INSTALL_PATH\=/System/Library/Extensions
    export DEFINES_MODULE\=NO
    export DEPLOYMENT_LOCATION\=NO
    export DEPLOYMENT_POSTPROCESSING\=NO
    export DEPLOYMENT_TARGET_CLANG_ENV_NAME\=IPHONEOS_DEPLOYMENT_TARGET
    export DEPLOYMENT_TARGET_CLANG_FLAG_NAME\=miphoneos-version-min
    export DEPLOYMENT_TARGET_CLANG_FLAG_PREFIX\=-miphoneos-version-min\=
    export DEPLOYMENT_TARGET_LD_ENV_NAME\=IPHONEOS_DEPLOYMENT_TARGET
    export DEPLOYMENT_TARGET_LD_FLAG_NAME\=ios_version_min
    export DEPLOYMENT_TARGET_SETTING_NAME\=IPHONEOS_DEPLOYMENT_TARGET
    export DEPLOYMENT_TARGET_SUGGESTED_VALUES\=9.0\ 9.2\ 10.0\ 10.2\ 11.0\ 11.2\ 11.4\ 12.1\ 12.3\ 13.0\ 13.2\ 13.4\ 13.6\ 14.1\ 14.3\ 14.5
    export DERIVED_FILES_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/DerivedSources
    export DERIVED_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/DerivedSources
    export DERIVED_SOURCES_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/DerivedSources
    export DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER\=NO
    export DEVELOPER_APPLICATIONS_DIR\=/Applications/Xcode.app/Contents/Developer/Applications
    export DEVELOPER_BIN_DIR\=/Applications/Xcode.app/Contents/Developer/usr/bin
    export DEVELOPER_DIR\=/Applications/Xcode.app/Contents/Developer
    export DEVELOPER_FRAMEWORKS_DIR\=/Applications/Xcode.app/Contents/Developer/Library/Frameworks
    export DEVELOPER_FRAMEWORKS_DIR_QUOTED\=/Applications/Xcode.app/Contents/Developer/Library/Frameworks
    export DEVELOPER_LIBRARY_DIR\=/Applications/Xcode.app/Contents/Developer/Library
    export DEVELOPER_SDK_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
    export DEVELOPER_TOOLS_DIR\=/Applications/Xcode.app/Contents/Developer/Tools
    export DEVELOPER_USR_DIR\=/Applications/Xcode.app/Contents/Developer/usr
    export DEVELOPMENT_LANGUAGE\=en
    export DONT_GENERATE_INFOPLIST_FILE\=NO
    export DO_HEADER_SCANNING_IN_JAM\=NO
    export DSTROOT\=/tmp/Pods.dst
    export DT_TOOLCHAIN_DIR\=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
    export DWARF_DSYM_FILE_NAME\=libFBReactNativeSpec.a.dSYM
    export DWARF_DSYM_FILE_SHOULD_ACCOMPANY_PRODUCT\=NO
    export DWARF_DSYM_FOLDER_PATH\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec
    export EFFECTIVE_PLATFORM_NAME\=-iphoneos
    export EMBEDDED_CONTENT_CONTAINS_SWIFT\=NO
    export EMBEDDED_PROFILE_NAME\=embedded.mobileprovision
    export EMBED_ASSET_PACKS_IN_PRODUCT_BUNDLE\=NO
    export ENABLE_BITCODE\=YES
    export ENABLE_DEFAULT_HEADER_SEARCH_PATHS\=YES
    export ENABLE_HARDENED_RUNTIME\=NO
    export ENABLE_HEADER_DEPENDENCIES\=YES
    export ENABLE_ON_DEMAND_RESOURCES\=NO
    export ENABLE_PREVIEWS\=NO
    export ENABLE_STRICT_OBJC_MSGSEND\=YES
    export ENABLE_TESTABILITY\=YES
    export ENABLE_TESTING_SEARCH_PATHS\=NO
    export ENTITLEMENTS_DESTINATION\=Signature
    export ENTITLEMENTS_REQUIRED\=YES
    export EXCLUDED_INSTALLSRC_SUBDIRECTORY_PATTERNS\=.DS_Store\ .svn\ .git\ .hg\ CVS
    export EXCLUDED_RECURSIVE_SEARCH_PATH_SUBDIRECTORIES\=\*.nib\ \*.lproj\ \*.framework\ \*.gch\ \*.xcode\*\ \*.xcassets\ \(\*\)\ .DS_Store\ CVS\ .svn\ .git\ .hg\ \*.pbproj\ \*.pbxproj
    export EXECUTABLE_EXTENSION\=a
    export EXECUTABLE_NAME\=libFBReactNativeSpec.a
    export EXECUTABLE_PATH\=libFBReactNativeSpec.a
    export EXECUTABLE_PREFIX\=lib
    export EXECUTABLE_SUFFIX\=.a
    export FILE_LIST\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects/LinkFileList
    export FIXED_FILES_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/FixedFiles
    export FRAMEWORK_FLAG_PREFIX\=-framework
    export FRAMEWORK_SEARCH_PATHS\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec\ 
    export FRAMEWORK_VERSION\=A
    export FULL_PRODUCT_NAME\=libFBReactNativeSpec.a
    export GCC3_VERSION\=3.3
    export GCC_C_LANGUAGE_STANDARD\=gnu11
    export GCC_DYNAMIC_NO_PIC\=NO
    export GCC_NO_COMMON_BLOCKS\=YES
    export GCC_OPTIMIZATION_LEVEL\=0
    export GCC_PFE_FILE_C_DIALECTS\=c\ objective-c\ c++\ objective-c++
    export GCC_PREFIX_HEADER\=Target\ Support\ Files/FBReactNativeSpec/FBReactNativeSpec-prefix.pch
    export GCC_PREPROCESSOR_DEFINITIONS\=POD_CONFIGURATION_DEBUG\=1\ DEBUG\=1\ \ COCOAPODS\=1
    export GCC_THUMB_SUPPORT\=YES
    export GCC_TREAT_WARNINGS_AS_ERRORS\=NO
    export GCC_VERSION\=com.apple.compilers.llvm.clang.1_0
    export GCC_VERSION_IDENTIFIER\=com_apple_compilers_llvm_clang_1_0
    export GCC_WARN_64_TO_32_BIT_CONVERSION\=YES
    export GCC_WARN_ABOUT_RETURN_TYPE\=YES_ERROR
    export GCC_WARN_UNDECLARED_SELECTOR\=YES
    export GCC_WARN_UNINITIALIZED_AUTOS\=YES_AGGRESSIVE
    export GCC_WARN_UNUSED_FUNCTION\=YES
    export GCC_WARN_UNUSED_VARIABLE\=YES
    export GENERATED_MODULEMAP_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/GeneratedModuleMaps-iphoneos
    export GENERATE_MASTER_OBJECT_FILE\=NO
    export GENERATE_PKGINFO_FILE\=NO
    export GENERATE_PROFILING_CODE\=NO
    export GENERATE_TEXT_BASED_STUBS\=NO
    export GID\=20
    export GROUP\=staff
    export HEADERMAP_INCLUDES_FLAT_ENTRIES_FOR_TARGET_BEING_BUILT\=YES
    export HEADERMAP_INCLUDES_FRAMEWORK_ENTRIES_FOR_ALL_PRODUCT_TYPES\=YES
    export HEADERMAP_INCLUDES_NONPUBLIC_NONPRIVATE_HEADERS\=YES
    export HEADERMAP_INCLUDES_PROJECT_HEADERS\=YES
    export HEADERMAP_USES_FRAMEWORK_PREFIX_ENTRIES\=YES
    export HEADERMAP_USES_VFS\=NO
    export HEADER_SEARCH_PATHS\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec/include\ \ \"/Users/will/dev/myapp/ios/Pods/Headers/Private\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Private/FBReactNativeSpec\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/DoubleConversion\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/FBLazyVector\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/FBReactNativeSpec\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/RCT-Folly\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/RCTRequired\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/RCTTypeSafety\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-Core\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-callinvoker\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-cxxreact\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-jsi\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-jsiexecutor\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-jsinspector\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-perflogger\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/React-runtimeexecutor\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/ReactCommon\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/Yoga\"\ \"/Users/will/dev/myapp/ios/Pods/Headers/Public/glog\"\ \"/Users/will/dev/myapp/ios/Pods/../../node_modules/react-native/React/FBReactNativeSpec/React/FBReactNativeSpec\"\ \"/Users/will/dev/myapp/ios/Pods/RCT-Folly\"
    export HIDE_BITCODE_SYMBOLS\=YES
    export HOME\=/Users/will
    export ICONV\=/usr/bin/iconv
    export INFOPLIST_EXPAND_BUILD_SETTINGS\=YES
    export INFOPLIST_OUTPUT_FORMAT\=binary
    export INFOPLIST_PREPROCESS\=NO
    export INLINE_PRIVATE_FRAMEWORKS\=NO
    export INSTALLHDRS_COPY_PHASE\=NO
    export INSTALLHDRS_SCRIPT_PHASE\=NO
    export INSTALL_DIR\=/tmp/Pods.dst/usr/local/lib
    export INSTALL_GROUP\=staff
    export INSTALL_MODE_FLAG\=u+w,go-w,a+rX
    export INSTALL_OWNER\=will
    export INSTALL_PATH\=/usr/local/lib
    export INSTALL_ROOT\=/tmp/Pods.dst
    export IPHONEOS_DEPLOYMENT_TARGET\=10.0
    export JAVAC_DEFAULT_FLAGS\=-J-Xms64m\ -J-XX:NewSize\=4M\ -J-Dfile.encoding\=UTF8
    export JAVA_APP_STUB\=/System/Library/Frameworks/JavaVM.framework/Resources/MacOS/JavaApplicationStub
    export JAVA_ARCHIVE_CLASSES\=YES
    export JAVA_ARCHIVE_TYPE\=JAR
    export JAVA_COMPILER\=/usr/bin/javac
    export JAVA_FRAMEWORK_RESOURCES_DIRS\=Resources
    export JAVA_JAR_FLAGS\=cv
    export JAVA_SOURCE_SUBDIR\=.
    export JAVA_USE_DEPENDENCIES\=YES
    export JAVA_ZIP_FLAGS\=-urg
    export JIKES_DEFAULT_FLAGS\=+E\ +OLDCSO
    export KASAN_DEFAULT_CFLAGS\=-DKASAN\=1\ -fsanitize\=address\ -mllvm\ -asan-globals-live-support\ -mllvm\ -asan-force-dynamic-shadow
    export KEEP_PRIVATE_EXTERNS\=NO
    export LD_DEPENDENCY_INFO_FILE\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects-normal/undefined_arch/FBReactNativeSpec_dependency_info.dat
    export LD_GENERATE_MAP_FILE\=NO
    export LD_MAP_FILE_PATH\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/FBReactNativeSpec-LinkMap-normal-undefined_arch.txt
    export LD_NO_PIE\=NO
    export LD_QUOTE_LINKER_ARGUMENTS_FOR_COMPILER_DRIVER\=YES
    export LEGACY_DEVELOPER_DIR\=/Applications/Xcode.app/Contents/PlugIns/Xcode3Core.ideplugin/Contents/SharedSupport/Developer
    export LEX\=lex
    export LIBRARY_DEXT_INSTALL_PATH\=/Library/DriverExtensions
    export LIBRARY_FLAG_NOSPACE\=YES
    export LIBRARY_FLAG_PREFIX\=-l
    export LIBRARY_KEXT_INSTALL_PATH\=/Library/Extensions
    export LIBRARY_SEARCH_PATHS\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec\ 
    export LINKER_DISPLAYS_MANGLED_NAMES\=NO
    export LINK_FILE_LIST_normal_arm64\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects-normal/arm64/FBReactNativeSpec.LinkFileList
    export LINK_WITH_STANDARD_LIBRARIES\=YES
    export LLVM_TARGET_TRIPLE_OS_VERSION\=ios10.0
    export LLVM_TARGET_TRIPLE_VENDOR\=apple
    export LOCALIZATION_EXPORT_SUPPORTED\=YES
    export LOCALIZED_STRING_MACRO_NAMES\=NSLocalizedString\ CFCopyLocalizedString
    export LOCALIZED_STRING_SWIFTUI_SUPPORT\=YES
    export LOCAL_ADMIN_APPS_DIR\=/Applications/Utilities
    export LOCAL_APPS_DIR\=/Applications
    export LOCAL_DEVELOPER_DIR\=/Library/Developer
    export LOCAL_LIBRARY_DIR\=/Library
    export LOCROOT\=/Users/will/dev/myapp/ios/Pods
    export LOCSYMROOT\=/Users/will/dev/myapp/ios/Pods
    export MACH_O_TYPE\=staticlib
    export MAC_OS_X_PRODUCT_BUILD_VERSION\=20E241
    export MAC_OS_X_VERSION_ACTUAL\=110301
    export MAC_OS_X_VERSION_MAJOR\=110000
    export MAC_OS_X_VERSION_MINOR\=110300
    export METAL_LIBRARY_FILE_BASE\=default
    export METAL_LIBRARY_OUTPUT_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec/
    export MODULE_CACHE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/ModuleCache.noindex
    export MTL_ENABLE_DEBUG_INFO\=INCLUDE_SOURCE
    export MTL_FAST_MATH\=YES
    export NATIVE_ARCH\=armv7
    export NATIVE_ARCH_32_BIT\=i386
    export NATIVE_ARCH_64_BIT\=x86_64
    export NATIVE_ARCH_ACTUAL\=x86_64
    export NO_COMMON\=YES
    export OBJECT_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects
    export OBJECT_FILE_DIR_normal\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects-normal
    export OBJROOT\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex
    export ONLY_ACTIVE_ARCH\=YES
    export OS\=MACOS
    export OSAC\=/usr/bin/osacompile
    export OTHER_CFLAGS\=\ -fmodule-map-file\=\"/Users/will/dev/myapp/ios/Pods/Headers/Public/React/React-Core.modulemap\"\ -fmodule-map-file\=\"/Users/will/dev/myapp/ios/Pods/Headers/Public/yoga/Yoga.modulemap\"
    export OTHER_CPLUSPLUSFLAGS\=\ -fmodule-map-file\=\"/Users/will/dev/myapp/ios/Pods/Headers/Public/React/React-Core.modulemap\"\ -fmodule-map-file\=\"/Users/will/dev/myapp/ios/Pods/Headers/Public/yoga/Yoga.modulemap\"
    export PACKAGE_TYPE\=com.apple.package-type.static-library
    export PASCAL_STRINGS\=YES
    export PATH\=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/local/bin:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/libexec:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/usr/bin:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/usr/local/bin:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/local/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
    export PATH_PREFIXES_EXCLUDED_FROM_HEADER_DEPENDENCIES\=/usr/include\ /usr/local/include\ /System/Library/Frameworks\ /System/Library/PrivateFrameworks\ /Applications/Xcode.app/Contents/Developer/Headers\ /Applications/Xcode.app/Contents/Developer/SDKs\ /Applications/Xcode.app/Contents/Developer/Platforms
    export PER_ARCH_OBJECT_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects-normal/undefined_arch
    export PER_VARIANT_OBJECT_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects-normal
    export PKGINFO_FILE_PATH\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/PkgInfo
    export PLATFORM_DEVELOPER_APPLICATIONS_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Applications
    export PLATFORM_DEVELOPER_BIN_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin
    export PLATFORM_DEVELOPER_LIBRARY_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library
    export PLATFORM_DEVELOPER_SDK_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs
    export PLATFORM_DEVELOPER_TOOLS_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Tools
    export PLATFORM_DEVELOPER_USR_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr
    export PLATFORM_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform
    export PLATFORM_DISPLAY_NAME\=iOS
    export PLATFORM_FAMILY_NAME\=iOS
    export PLATFORM_NAME\=iphoneos
    export PLATFORM_PREFERRED_ARCH\=arm64
    export PLATFORM_PRODUCT_BUILD_VERSION\=18E182
    export PLIST_FILE_OUTPUT_FORMAT\=binary
    export PODS_BUILD_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products
    export PODS_CONFIGURATION_BUILD_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos
    export PODS_ROOT\=/Users/will/dev/myapp/ios/Pods
    export PODS_TARGET_SRCROOT\=/Users/will/dev/myapp/ios/Pods/../../node_modules/react-native/React/FBReactNativeSpec
    export PODS_XCFRAMEWORKS_BUILD_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/XCFrameworkIntermediates
    export PRECOMPS_INCLUDE_HEADERS_FROM_BUILT_PRODUCTS_DIR\=YES
    export PRECOMP_DESTINATION_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/PrefixHeaders
    export PRESERVE_DEAD_CODE_INITS_AND_TERMS\=NO
    export PRODUCT_BUNDLE_IDENTIFIER\=org.cocoapods.FBReactNativeSpec
    export PRODUCT_MODULE_NAME\=FBReactNativeSpec
    export PRODUCT_NAME\=FBReactNativeSpec
    export PRODUCT_SETTINGS_PATH\=
    export PRODUCT_TYPE\=com.apple.product-type.library.static
    export PROFILING_CODE\=NO
    export PROJECT\=Pods
    export PROJECT_DERIVED_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/DerivedSources
    export PROJECT_DIR\=/Users/will/dev/myapp/ios/Pods
    export PROJECT_FILE_PATH\=/Users/will/dev/myapp/ios/Pods/Pods.xcodeproj
    export PROJECT_NAME\=Pods
    export PROJECT_TEMP_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build
    export PROJECT_TEMP_ROOT\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex
    export RECURSIVE_SEARCH_PATHS_FOLLOW_SYMLINKS\=YES
    export REMOVE_CVS_FROM_RESOURCES\=YES
    export REMOVE_GIT_FROM_RESOURCES\=YES
    export REMOVE_HEADERS_FROM_EMBEDDED_BUNDLES\=YES
    export REMOVE_HG_FROM_RESOURCES\=YES
    export REMOVE_SVN_FROM_RESOURCES\=YES
    export REZ_COLLECTOR_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/ResourceManagerResources
    export REZ_EXECUTABLE\=YES
    export REZ_OBJECTS_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/ResourceManagerResources/Objects
    export REZ_SEARCH_PATHS\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec\ 
    export SCAN_ALL_SOURCE_FILES_FOR_INCLUDES\=NO
    export SCRIPT_INPUT_FILE_0\=/Users/will/dev/myapp/node_modules/react-native/Libraries
    export SCRIPT_INPUT_FILE_COUNT\=1
    export SCRIPT_INPUT_FILE_LIST_COUNT\=0
    export SCRIPT_OUTPUT_FILE_0\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/DerivedSources/codegen-FBReactNativeSpec.log
    export SCRIPT_OUTPUT_FILE_1\=/Users/will/dev/myapp/node_modules/react-native/React/FBReactNativeSpec/FBReactNativeSpec/FBReactNativeSpec.h
    export SCRIPT_OUTPUT_FILE_2\=/Users/will/dev/myapp/node_modules/react-native/React/FBReactNativeSpec/FBReactNativeSpec/FBReactNativeSpec-generated.mm
    export SCRIPT_OUTPUT_FILE_COUNT\=3
    export SCRIPT_OUTPUT_FILE_LIST_COUNT\=0
    export SDKROOT\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk
    export SDK_DIR\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk
    export SDK_DIR_iphoneos\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk
    export SDK_DIR_iphoneos14_5\=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk
    export SDK_NAME\=iphoneos14.5
    export SDK_NAMES\=iphoneos14.5
    export SDK_PRODUCT_BUILD_VERSION\=18E182
    export SDK_VERSION\=14.5
    export SDK_VERSION_ACTUAL\=140500
    export SDK_VERSION_MAJOR\=140000
    export SDK_VERSION_MINOR\=140500
    export SED\=/usr/bin/sed
    export SEPARATE_STRIP\=NO
    export SEPARATE_SYMBOL_EDIT\=NO
    export SET_DIR_MODE_OWNER_GROUP\=YES
    export SET_FILE_MODE_OWNER_GROUP\=NO
    export SHALLOW_BUNDLE\=NO
    export SHARED_DERIVED_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec/DerivedSources
    export SHARED_PRECOMPS_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/PrecompiledHeaders
    export SKIP_INSTALL\=YES
    export SOURCE_ROOT\=/Users/will/dev/myapp/ios/Pods
    export SRCROOT\=/Users/will/dev/myapp/ios/Pods
    export STRINGS_FILE_INFOPLIST_RENAME\=YES
    export STRINGS_FILE_OUTPUT_ENCODING\=binary
    export STRIP_BITCODE_FROM_COPIED_FILES\=YES
    export STRIP_INSTALLED_PRODUCT\=NO
    export STRIP_STYLE\=debugging
    export STRIP_SWIFT_SYMBOLS\=YES
    export SUPPORTED_DEVICE_FAMILIES\=1,2
    export SUPPORTED_PLATFORMS\=iphoneos\ iphonesimulator
    export SUPPORTS_MACCATALYST\=YES
    export SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD\=YES
    export SUPPORTS_TEXT_BASED_API\=NO
    export SWIFT_ACTIVE_COMPILATION_CONDITIONS\=DEBUG\ 
    export SWIFT_OPTIMIZATION_LEVEL\=-Onone
    export SWIFT_PLATFORM_TARGET_PREFIX\=ios
    export SWIFT_RESPONSE_FILE_PATH_normal_arm64\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Objects-normal/arm64/FBReactNativeSpec.SwiftFileList
    export SWIFT_VERSION\=5.0
    export SYMROOT\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products
    export SYSTEM_ADMIN_APPS_DIR\=/Applications/Utilities
    export SYSTEM_APPS_DIR\=/Applications
    export SYSTEM_CORE_SERVICES_DIR\=/System/Library/CoreServices
    export SYSTEM_DEMOS_DIR\=/Applications/Extras
    export SYSTEM_DEVELOPER_APPS_DIR\=/Applications/Xcode.app/Contents/Developer/Applications
    export SYSTEM_DEVELOPER_BIN_DIR\=/Applications/Xcode.app/Contents/Developer/usr/bin
    export SYSTEM_DEVELOPER_DEMOS_DIR\=/Applications/Xcode.app/Contents/Developer/Applications/Utilities/Built\ Examples
    export SYSTEM_DEVELOPER_DIR\=/Applications/Xcode.app/Contents/Developer
    export SYSTEM_DEVELOPER_DOC_DIR\=/Applications/Xcode.app/Contents/Developer/ADC\ Reference\ Library
    export SYSTEM_DEVELOPER_GRAPHICS_TOOLS_DIR\=/Applications/Xcode.app/Contents/Developer/Applications/Graphics\ Tools
    export SYSTEM_DEVELOPER_JAVA_TOOLS_DIR\=/Applications/Xcode.app/Contents/Developer/Applications/Java\ Tools
    export SYSTEM_DEVELOPER_PERFORMANCE_TOOLS_DIR\=/Applications/Xcode.app/Contents/Developer/Applications/Performance\ Tools
    export SYSTEM_DEVELOPER_RELEASENOTES_DIR\=/Applications/Xcode.app/Contents/Developer/ADC\ Reference\ Library/releasenotes
    export SYSTEM_DEVELOPER_TOOLS\=/Applications/Xcode.app/Contents/Developer/Tools
    export SYSTEM_DEVELOPER_TOOLS_DOC_DIR\=/Applications/Xcode.app/Contents/Developer/ADC\ Reference\ Library/documentation/DeveloperTools
    export SYSTEM_DEVELOPER_TOOLS_RELEASENOTES_DIR\=/Applications/Xcode.app/Contents/Developer/ADC\ Reference\ Library/releasenotes/DeveloperTools
    export SYSTEM_DEVELOPER_USR_DIR\=/Applications/Xcode.app/Contents/Developer/usr
    export SYSTEM_DEVELOPER_UTILITIES_DIR\=/Applications/Xcode.app/Contents/Developer/Applications/Utilities
    export SYSTEM_DEXT_INSTALL_PATH\=/System/Library/DriverExtensions
    export SYSTEM_DOCUMENTATION_DIR\=/Library/Documentation
    export SYSTEM_KEXT_INSTALL_PATH\=/System/Library/Extensions
    export SYSTEM_LIBRARY_DIR\=/System/Library
    export TAPI_VERIFY_MODE\=ErrorsOnly
    export TARGETED_DEVICE_FAMILY\=1,2
    export TARGETNAME\=FBReactNativeSpec
    export TARGET_BUILD_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Products/Debug-iphoneos/FBReactNativeSpec
    export TARGET_DEVICE_IDENTIFIER\=b6effe14dbd383824b0abaa6ee6836df89faba21
    export TARGET_DEVICE_MODEL\=iPhone10,1
    export TARGET_DEVICE_OS_VERSION\=14.4.2
    export TARGET_DEVICE_PLATFORM_NAME\=iphoneos
    export TARGET_NAME\=FBReactNativeSpec
    export TARGET_TEMP_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build
    export TEMP_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build
    export TEMP_FILES_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build
    export TEMP_FILE_DIR\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build
    export TEMP_ROOT\=/Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex
    export TEST_FRAMEWORK_SEARCH_PATHS\=\ /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks\ /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk/Developer/Library/Frameworks
    export TEST_LIBRARY_SEARCH_PATHS\=\ /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/lib
    export TOOLCHAINS\=com.apple.dt.toolchain.XcodeDefault
    export TOOLCHAIN_DIR\=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
    export TREAT_MISSING_BASELINES_AS_TEST_FAILURES\=NO
    export UID\=501
    export UNSTRIPPED_PRODUCT\=NO
    export USER\=will
    export USER_APPS_DIR\=/Users/will/Applications
    export USER_LIBRARY_DIR\=/Users/will/Library
    export USE_DYNAMIC_NO_PIC\=YES
    export USE_HEADERMAP\=YES
    export USE_HEADER_SYMLINKS\=NO
    export USE_LLVM_TARGET_TRIPLES\=YES
    export USE_LLVM_TARGET_TRIPLES_FOR_CLANG\=YES
    export USE_LLVM_TARGET_TRIPLES_FOR_LD\=YES
    export USE_LLVM_TARGET_TRIPLES_FOR_TAPI\=YES
    export USE_RECURSIVE_SCRIPT_INPUTS_IN_SCRIPT_PHASES\=YES
    export VALIDATE_DEVELOPMENT_ASSET_PATHS\=YES_ERROR
    export VALIDATE_PRODUCT\=NO
    export VALIDATE_WORKSPACE\=YES_ERROR
    export VALID_ARCHS\=arm64\ arm64e\ armv7\ armv7s
    export VERBOSE_PBXCP\=NO
    export VERSION_INFO_BUILDER\=will
    export VERSION_INFO_FILE\=FBReactNativeSpec_vers.c
    export VERSION_INFO_STRING\=\"@\(\#\)PROGRAM:FBReactNativeSpec\ \ PROJECT:Pods-\"
    export WRAP_ASSET_PACKS_IN_SEPARATE_DIRECTORIES\=NO
    export XCODE_APP_SUPPORT_DIR\=/Applications/Xcode.app/Contents/Developer/Library/Xcode
    export XCODE_PRODUCT_BUILD_VERSION\=12E262
    export XCODE_VERSION_ACTUAL\=1250
    export XCODE_VERSION_MAJOR\=1200
    export XCODE_VERSION_MINOR\=1250
    export XPCSERVICES_FOLDER_PATH\=/XPCServices
    export YACC\=yacc
    export arch\=undefined_arch
    export variant\=normal
    /bin/sh -c /Users/will/Library/Developer/Xcode/DerivedData/myapp-dloomuvupznyfccpvgvmgftrjffb/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/FBReactNativeSpec.build/Script-B4E323DB11FD523AF1F108EDFA5396CE.sh

Command PhaseScriptExecution failed with a nonzero exit code
```

**Comment by levymetal** [comment]
@williamhaley's fix pointed me in the right direction. Looking into the `nodenv: cannot rehash` error, I found that `nodenv-rehash` will blow up when run in parallel: https://github.com/nodenv/nodenv/blob/master/libexec/nodenv-rehash. 

Naturally, I thought `find-node.sh` may have being running in parallel. But after some investigation this doesn't appear to be case. Then I checked the `~/.nodenv/shims` directory and found I had a rogue `.nodenv-shim` file. Indeed, I wasn't even able to run `nodenv rehash` manually. After deleting this file, everything works properly - no need to patch `find-node.sh`.

**Comment by gtebbutt** [comment]
I was seeing the same issue on a new project created with RN 0.64.1. Explicitly running `pod install` from the `ios` folder altered the `FBReactNativeSpec` checksum in `Podfile.lock`, and everything built without an issue after that.

**Comment by bcsaxon** [comment]
I ONLY commented out `project/node_modules/react-native/scripts/find-node.sh` like @sumanbhattarai mentioned as the second part of his solution above. I'm still scared of bash a bit so I didn't dig in to mess with my paths. 

Think this will cause me headaches in the future? hopefully, a true fix and less of a patch will be coming soon.

**Comment by levymetal** [comment]
@hlynbech shims are an `[x]env` concept (`rbenv/pyenv/nodenv` etc) so you wouldn't have this exact issue with `nvm`. Have you read over the comments on https://github.com/facebook/react-native/issues/31181? A lot of `nvm` users there with various fixes.

Also, even if you aren't actively using `nodenv`, do you have it installed (`nodenv -v`)? If you've got it installed and there's a `.nodenv-shim` file in the shims directory, you'll get this error because `find-node.sh` will attempt an `eval "$(nodenv init -)"`.

**Comment by lincolnBerlick** [comment]
i had the same problem probably was any path variable bad configured with mvn and node, i got resolve, removing nvm, node and reinstall all
i followed this tutorial
https://tecadmin.net/install-nvm-macos-with-homebrew/

be attention with the node version that you need,

good luck

**Comment by ikejs** [comment]
SOLVED
I had this problem while using oh-my-zsh. I disabled zsh and switched my terminal to **bash**, and it worked.
Might have to reinstall dependencies/pods.

**Comment by Vanilna** [comment]
For me, none of the above solutions haven't worked :( 
The problem appeared after installing nvm and only removing nvm entirely helped.

**Comment by ikejs** [comment]
I fixed this by making sure my bash prefix is set to something like `/Users/name/...` instead of `/usr/bin/...`

**Comment by wuweikd** [comment]
I finally solved it, I updated my node (not in nvm). And I made sure that the node in my nvm is the last version

**Comment by akashkumarpu** [comment]
> Removing
> 
> ```
> export NVM_DIR="$HOME/.nvm"
> [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
> [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
> ```
> 
> from `~/.bash_profile` .
> 
> AND
> 
> Commenting out all the code from `project/node_modules/react-native/scripts/find-node.sh`
> helped to solve this issue.

I am also facing the same issue and when I tried to open .bash_profile it is empty

**Comment by jamesguan** [comment]
I had this issue when I was running node v15. I installed lts with nvm without messing with nvm and it built properly and ran.

**Comment by akashkumarpu** [comment]
Try removing the spaces on the project path and run it

It resolved the problem for me

**Comment by alexleone** [comment]
So the issue for me was related to how I setup NVM to rewrite the cd command to switch node versions based on the .nvmrc file. When I removed the cdnvm command as seen here https://github.com/nvm-sh/nvm#automatically-call-nvm-use then it all worked. Hopefully, this helps someone.

**Comment by nidongara** [comment]
So the issue for me was because my Jenkins worker was still on bash. I moved it to zsh. Deleted .nvm for safety(not sure if it made a difference). This seemed to work for me and was able to successfully build with fastlane too.

**Comment by ikejs** [comment]
I solved this by using [fnm](https://github.com/Schniz/fnm) instead of nvm

**Comment by ibra4** [comment]
Same Issue on Microsoft App Center

**Comment by sayurimizuguchi** [comment]
had this issue using Fastlane on our CI server, in order to to fix it the issue, I added some commands to clean pods entirely and do a complete clean installation - before run the Fastlane bundle commands

```
rm -rf ~/Library/Developer/Xcode/DerivedData/*
rm -rf Pods
pod install --clean-install
// ... bundle exec fastlane ios
```

It was using cache from previous pods/builds. If you're having this issue you can try these commands above.

**Comment by gaviin-h** [comment]
I know there seem to be some solutions here but none worked for me. The solution that ended up working was really simple. Go through the absolute path of your project, and make sure there are no spaces! I had to edit a file in a directory 3 parents prior to my app folder.  

For me it was something like "users/.../Desktop/school/class abc/test/project" <- I had to make it 'class-abc'

**Comment by parthkanani93** [comment]
> Removing
> 
> ```
> export NVM_DIR="$HOME/.nvm"
> [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
> [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
> ```
> 
> from `~/.bash_profile` .
> 
> AND
> 
> Commenting out all the code from `project/node_modules/react-native/scripts/find-node.sh` helped to solve this issue.

Thanks this works for me

**Comment by vladakg85** [comment]
I am on react 0.67.2 (new project) I tried on two different machines with this issue.
Only @sumanbhattarai solution works.

**Comment by AbhilashG97** [comment]
> I solved this by using [fnm](https://github.com/Schniz/fnm) instead of nvm

I had the same issue, completely removing nvm and using [fnm](https://github.com/Schniz/fnm) solved this issue for me. Thank you @ikejs

**Comment by gonglixin980926** [comment]
I had the same problem, but I fixed it using `nvm Install v16.10.0`

**Comment by adityadua24** [comment]
> Removing
> 
> ```
> export NVM_DIR="$HOME/.nvm"
> [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
> [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
> ```
> 
> from `~/.bash_profile` .
> 
> AND
> 
> Commenting out all the code from `project/node_modules/react-native/scripts/find-node.sh` helped to solve this issue.



> Removing
> 
> ```
> export NVM_DIR="$HOME/.nvm"
> [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
> [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
> ```
> 
> from `~/.bash_profile` .
> 
> AND
> 
> Commenting out all the code from `project/node_modules/react-native/scripts/find-node.sh` helped to solve this issue.

This worked for me. May i ask how did u debug this and come to this conclusion ? @sumanbhattarai

**Comment by mineturtle2137** [comment]
> Removing
> 
> ```
> export NVM_DIR="$HOME/.nvm"
> [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
> [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
> ```
> 
> from `~/.bash_profile` .
> 
> AND
> 
> Commenting out all the code from `project/node_modules/react-native/scripts/find-node.sh` helped to solve this issue.

I had the same issue, problem details are clear in the XCode:
<img width="547" alt="image" src="https://user-images.githubusercontent.com/80891940/155496379-d34a26e6-a2ce-45a6-8d75-cf119bfe1202.png">
and it seems the problem in the script is here:
<img width="385" alt="image" src="https://user-images.githubusercontent.com/80891940/155495892-4f408c96-86f3-4b9e-9bc8-dfb62c577d79.png">


Try to fix the nvm on your computer, as commenting out the node_modules makes no sense.
Usually, the problem itself is the nvm, so try to check if the nvm has node and set the default alias:

Try:
`nvm alias default node`

Or install the node by nvm first:
```
nvm install node
nvm alias default node
```

**Comment by SudoPlz** [comment]
@mineturtle2137 's solution solved it for me, thank you so much man!! 🙏 🙏 🙏 🙏 🙏 🎉 🎉 🎉

**Comment by vasiminamdar** [comment]
Below solution fixed issues for me.

Commenting out all references for NVM_DIR in project_repository/node_modules/react-native/scripts/find-node.sh and ~/.bash_profile fixed the issue. 

Sample code 

```
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  
```
